test/blas.cpp

```diff
@@ -245,3 +245,23 @@ TYPED_TEST(MatrixMultiply, MultiGPURectangleVector_CPP)
 }
 
 #undef DEVICE_ITERATE
+
+TEST(MatrixMultiply, ISSUE_1882)
+{
+    const int m = 2;
+    const int n = 3;
+    af::array A = af::randu(m, n);
+    af::array BB = af::randu(n, m);
+    af::array B = BB(0, af::span);
+
+    af::array res1 = af::matmul(A.T(), B.T());
+    af::array res2 = af::matmulTT(A, B);
+
+    std::vector<float> hres1(res1.elements());
+    std::vector<float> hres2(res2.elements());
+    ASSERT_EQ(hres1.size(), hres2.size());
+
+    for (size_t i = 0; i < hres1.size(); i++) {
+        ASSERT_EQ(hres1[i], hres2[i]);
```

Comment:
I believe that `hres1` and `hres2` are just vectors full of zeros, since no data were copied into them. So this assert will always be true.